src/ApplicationSampler.cpp

```diff
@@ -86,7 +92,15 @@ namespace geopm
     }
 
     ApplicationSamplerImp::ApplicationSamplerImp()
+        : ApplicationSamplerImp(environment().record_filter())
+    {
+
+    }
+
+    ApplicationSamplerImp::ApplicationSamplerImp(const std::string &filter_name)
```

Comment:
it might be easier to test the passthrough if RecordFilters are injected instead of the name

Reply:
We need a way to generate record filters for each process after construction.  That is facilitated with a string now.  I'm not sure how we want to mock out the RecordFilter::make_unique() call, which is where we will run into testing problems.

Reply:
maybe:

```
ApplicationSamplerImp(std::function<std::unique_ptr<RecordFilter>()> filter_factory);
```